File: fc.java
// fc.java
// Java native implementation of FileConnect (fc) loading logic

import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class fc {
    public static String linkText(String filepath) throws Exception {
        return new String(Files.readAllBytes(Paths.get(filepath)));
    }

    public static String linkRun(String filepath, String... args) throws Exception {
        String[] cmd = new String[args.length + 1];
        cmd[0] = filepath;
        System.arraycopy(args, 0, cmd, 1, args.length);

        ProcessBuilder pb = new ProcessBuilder(cmd);
        Process process = pb.start();

        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
        StringBuilder builder = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            builder.append(line).append("\n");
        }

        int exitCode = process.waitFor();
        if (exitCode != 0) {
            throw new RuntimeException("Process failed with exit code: " + exitCode);
        }

        return builder.toString();
    }
}

File: fc.cs
// fc.cs
// C# native implementation of FileConnect (fc) loading logic

using System;
using System.IO;
using System.Diagnostics;

public static class fc {
    public static string LinkText(string filepath) {
        if (!File.Exists(filepath)) {
            throw new FileNotFoundException("FileConnect: file not found: " + filepath);
        }
        return File.ReadAllText(filepath);
    }

    public static string LinkRun(string filepath, string args) {
        ProcessStartInfo startInfo = new ProcessStartInfo() {
            FileName = filepath,
            Arguments = args,
            RedirectStandardOutput = true,
            RedirectStandardError = true,
            UseShellExecute = false,
            CreateNoWindow = true
        };

        using (Process process = Process.Start(startInfo)) {
            using (StreamReader reader = process.StandardOutput) {
                string result = reader.ReadToEnd();
                process.WaitForExit();
                if (process.ExitCode != 0) {
                    string error = process.StandardError.ReadToEnd();
                    throw new Exception("Execution failed: " + error);
                }
                return result;
            }
        }
    }
}
